### ISSUE TITLE
Execute code on enter/exit fence?

### ISSUE THREAD

**Issue by dylanvdmerwe** [cowbell/cordova-plugin-geofence#35]
Is it possible to run some code when a user enters or exists a fence? For example I want to call a REST method which lets the server know what has happened. This can then be used for really cool things such as sending a push message to the user, etc etc. 

For example:

window.geofence.addOrUpdate({
    id: "69ca1b88-6fbe-4e80-a4d4-ff4d3748acdb",
    latitude: 50.2980049,
    longitude: 18.6593152,
    radius: 3000,
    transitionType: TransitionType.ENTER,
    notification: {
        id: 1,
        title: "Welcome in Gliwice",
        text: "You just arrived to Gliwice city center.",
        openAppOnClick: true
    },
    action: function(id, transitionType) {
        // perform some code such as as calling a REST web service
    }
}).then(function() {
    console.log('Geofence successfully added');
}, function(reason) {
    console.log('Adding geofence failed', reason);
})

**Comment by tsubik** [comment]
Only by `receiveTransition` event so far. Take a look [here](https://github.com/cowbell/ionic-geofence/blob/master/www/js/app.js#L45). It's not gonna work when your app is closed though.

**Comment by dylanvdmerwe** [comment]
There must be a way to get a customization action to execute when a geofence event happens though? Natively this can happen on both iOS and Android as far as I know.

**Comment by jgcaruso** [comment]
I issued a pull request that was merged a few days ago that allows this for iOS. It uses NotificationCenter to send a message to anyone who is registered to listen to it. Check it out, there are details on how to implement it (Objective-c code only, but Swift shouldn't be too hard to figure out): https://github.com/cowbell/cordova-plugin-geofence/pull/43

I'll be working on Android support for this in a little while.

**Comment by tsubik** [comment]
Example how to do this in native code is [here](https://github.com/cowbell/cordova-plugin-geofence#listening-for-geofence-transitions-in-native-code). I would say this is similar to #53 and #86.

**Comment by johnrobertcobbold** [comment]
This is already do-able with this plugin (at least on iOS, haven't started working on Android yet).

**Comment by frmi** [comment]
I have tested it on Android and it seems to work aswell without native code

**Comment by razagill** [comment]
@frmi Can you post your code? It isn't working form me.

**Comment by frmi** [comment]
@razagill i'm sorry, i was wrong this does not work if the app is forcefully closed, since the webview will be null.

**Comment by johnrobertcobbold** [comment]
@razagill "since the webview will be null" - is that the reason on Android ? When the app was forcefully closed, iOS re-opens the app on a geofence event and therefore recreates the webview .

**Comment by razagill** [comment]
@johnrobertcobbold Not exactly, you can't run javascript code in the background on android. So you have to modify the plugin and add new class that extends the BroadcastReceiver (details are on the main page of this plugin), I was able to catch the transitions in the background after this modification.

**Comment by johnrobertcobbold** [comment]
So basically like iOS using the new wkWebView... oh great :)

**Comment by frmi** [comment]
@razagill @johnrobertcobbold The plugin is still alive even if you forcefully kill the app. The notification is successfully tiggered, but the onTransitionReceived callback cannot be executed since the cordova web view has been released by closing the app.
`02-24 12:24:49.309      711-972/com.frmi.app D/GeofencePlugin﹕ Transition Event Received!
02-24 12:24:49.309      711-972/com.frmi.app D/GeofencePlugin﹕ Webview is null`

I guess there is no way around the broadcast receiver for now.

**Comment by Wade-BuildOtto** [comment]
their must be a method to call a custom url scheme, rather than setting up a notification, it seems like you have to setup everything you need to when you create the fence and the system takes over. 

maybe use it in conjunction with https://github.com/EddyVerbruggen/Custom-URL-scheme
then capture the data from the url and execute the code.
